src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -1270,17 +1330,41 @@ impl AccessChecker for PrivilegeAccess {
             }
             Plan::Commit => {}
             Plan::Abort => {}
-            Plan::ShowWarehouses => {}
-            Plan::ShowOnlineNodes => {}
-            Plan::DropWarehouse(_) => {}
-            Plan::ResumeWarehouse(_) => {}
-            Plan::SuspendWarehouse(_) => {}
-            Plan::RenameWarehouse(_) => {}
-            Plan::InspectWarehouse(_) => {}
-            Plan::DropWarehouseCluster(_) => {}
-            Plan::RenameWarehouseCluster(_) => {}
-            Plan::UseWarehouse(_) => {}
-            Plan::CreateWarehouse(_) => {}
+            Plan::ShowWarehouses => {
+                // check privilege in interpreter
+            }
+            Plan::ShowOnlineNodes => {
+                // todo: now no limit
+            }
+            Plan::DropWarehouse(plan) => {
+                self.validate_warehouse_ownership(plan.warehouse.clone(), identity).await.transpose()?;
+            }
+            Plan::ResumeWarehouse(plan) => {
+                self.validate_warehouse_ownership(plan.warehouse.clone(), identity).await.transpose()?;
+            }
+            Plan::SuspendWarehouse(plan) => {
+                self.validate_warehouse_ownership(plan.warehouse.clone(), identity).await.transpose()?;
+            }
+            Plan::RenameWarehouse(plan) => {
+                self.validate_warehouse_ownership(plan.warehouse.clone(), identity).await.transpose()?;
+            }
+            Plan::InspectWarehouse(plan) => {
+                self.validate_warehouse_ownership(plan.warehouse.clone(), identity).await.transpose()?;
+            }
+            Plan::DropWarehouseCluster(plan) => {
+                self.validate_warehouse_ownership(plan.warehouse.clone(), identity).await.transpose()?;
+            }
+            Plan::RenameWarehouseCluster(plan) => {
+                self.validate_warehouse_ownership(plan.warehouse.clone(), identity).await.transpose()?;
+            }
+            Plan::UseWarehouse(plan) => {
+                self.validate_warehouse_ownership(plan.warehouse.clone(), identity).await.transpose()?;
+            }
+            Plan::CreateWarehouse(_) => {
+                // only current role has global level create warehouse privilege, it will pass
```

Comment:
Maybe only support_forward_warehouse_request need to check create warehouse privilege? cc @flaneur2020 @zhang2014